### PR TITLE
separate storage chart form clusterpedia

### PR DESCRIPTION
**Pull request by calvin0327**
separate storage chart form clusterpedia.

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/clusterpedia/clusterpedia-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/clusterpedia/clusterpedia-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/clusterpedia/clusterpedia-helm/blob/main/CONTRIBUTING.md).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

File: charts/clusterpedia-core/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+#
+# Library charts provide useful utilities or functions for the chart developer. They're included as
+# a dependency of application charts to inject those utilities and functions into the rendering
+# pipeline. Library charts do not define any templates and therefore cannot be deployed.
+type: application
+
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 1.0.3
```
Comment (Iceber) [inline charts/clusterpedia-core/Chart.yaml:18]: iFor the new chart we iterated from 0.1

File: charts/clusterpedia-core/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+# It is recommended to use it with quotes.
+appVersion: "v0.5.1"
```
Comment (Iceber) [inline charts/clusterpedia-core/Chart.yaml:24]: 0.5.x does not support the storage layer plugin, we can set it to 0.6.0 first

File: charts/clusterpedia-core/README.md [hunk 3]
```diff
@@ -0,0 +1,245 @@
+# Clusterpedia
```
Comment (Iceber) [inline charts/clusterpedia-core/README.md:1]: We also need to update the readme, which can be done in a new pr

File: charts/clusterpedia-core/templates/apiserver-deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    {{- if .Values.apiserver.podAnnotations }}
+      annotations: {{- include "common.tplvalues.render" (dict "value" .Values.apiserver.podAnnotations "context" $) | nindent 8 }}
+    {{- end }}
+      labels: {{ include "common.labels.standard" . | nindent 8 }}
+        app: {{ include "clusterpedia.apiserver.fullname" . }}
+      {{- if .Values.podLabels }}
+        {{- include "common.tplvalues.render" (dict "value" .Values.apiserver.podLabels "context" $) | nindent 8 }}
+      {{- end }}
+    spec: {{ include "clusterpedia.apiserver.imagePullSecrets" . | nindent 6 }}
+    {{- if (include "clusterpedia.storage.initContainer" .) -}}
```
Comment (Iceber) [inline charts/clusterpedia-core/templates/apiserver-deployment.yaml:30]: s/initContainer/initContainers

File: charts/clusterpedia-core/templates/apiserver-deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+      containers:
+      - name: {{ include "clusterpedia.apiserver.fullname" . }}
+        image: {{ template "clusterpedia.apiserver.image" . }}
+        imagePullPolicy: {{ .Values.apiserver.image.pullPolicy }}
+        command:
+        - /usr/local/bin/apiserver
+        - --secure-port=443
+        - --storage-config=/etc/clusterpedia/storage/storage-config.yaml
+        - -v=3
+      {{- if .Values.storage.name }}
```
Comment (Iceber) [inline charts/clusterpedia-core/templates/apiserver-deployment.yaml:42]: .Values.storage.name is not in values.yaml

File: charts/clusterpedia-core/values.yaml [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+  imageRegistry: ""
+  ## E.g.
+  ## imagePullSecrets:
+  ##   - myRegistryKeySecretName
+  imagePullSecrets: []
+
+## @param storageInstallMode "internal" and "external" are provided
+## "internal" means automatic Database installation, default mysql
+## "external" means to install the database yourself, must specify the database connection parameters
+storageInstallMode: "internal"
```
Comment (Iceber) [inline charts/clusterpedia-core/values.yaml:17]: This field is no longer needed

File: charts/internal-storage/values.yaml [hunk 7]
```diff
@@ -0,0 +1,413 @@
...
+  password: ""
+  ## @param externalStorage.database Name for a custom database
+  ##
+  database: ""
+  ## @param externalStorage.createDatabase whether to create the .Values.externalStorage.database or not
+  ##
+  createDatabase: false
+
+## clusterpedia apiserver config
+apiserver:
```
Comment (Iceber) [inline charts/internal-storage/values.yaml:107]: Contains many fields that are no longer needed

File: charts/internal-storage/Chart.yaml [hunk 8]
```diff
@@ -0,0 +1,49 @@
+apiVersion: v2
+name: clusterpedia-internal-storage
```
Comment (Iceber) [inline charts/internal-storage/Chart.yaml:2]: The storage layer name is internalstorage
s/internal-storage/internalstorage

**Comment by calvin0327** [comment]
now, we separate mysql and pgresql from internalstorage. so I will close it.

**Pull request closed without merging**